(DOCSP-25302) Tutorial for creating ephemeral project and cluster (#164)
mongodb/docs-atlas-cli: source/atlas-cli-ephemeral-cluster.txt

.. _atlas-cli-ephemeral-cluster:

=====================================================
Test Automations with Ephemeral Projects and Clusters
=====================================================

.. default-domain:: mongodb

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: singlecol

This tutorial demonstrates how to quickly create and delete ephemeral 

> |service| projects and {+clusters+} using the {+atlas-cli+}. Ephemeral

Reviewer: I was just going to ask if we could use a simpler word than ephemeral, but explaining it here is a nice touch

projects and {+clusters+} provide temporary testing environments that

> are isolated from your production projects and {+clusters+}. You can use

Reviewer: avoid passive voice
```suggestion
you isolate from your production projects and {+clusters+}. You can use
```

ephemeral projects and {+clusters+} to test your automation scripts
before using the scripts on production {+clusters+}.

.. _atlas-cli-ephemeral-reqs:

Prerequisites 
-------------

Before you begin, complete the following tasks: 

> - Create an |service| user account

Reviewer: ```suggestion
- Create an |service| user account.
```

- :ref:`Create <atlas-organizations-create>` an 
  :atlas:`Atlas organization </organizations-projects/>` and

>   note its organization ID. You can find the organization ID by running the :ref:`atlas-organizations-list` command.

Reviewer: question: do you think it makes sense for us to add an explicit step below that shows users how to get their org id or do you think it should stay in the prereqs?

- :ref:`install-atlas-cli`.
- :ref:`connect-atlas-cli`.

Follow These Steps
------------------

.. procedure::
   :style: normal

   .. step:: Create the ephemeral project, {+cluster+}, and database user.

      a. Run the :ref:`atlas-users-describe` command to find your 

>          |service| user ID. Replace <YOUR-EMAIL> with the email address

Reviewer: question: is this returned in a user ID field? Might be worth showing example output

         that is associated with your |service| user account.

         .. code-block::

            atlas users describe --username <YOUR-EMAIL>
      
      b. Run the :ref:`atlas-projects-create` command to create the
         ephemeral project. Replace <ORG-ID> with the organization ID
         and replace <YOUR-USER-ID> with your |service| user ID that
         you retrieved in the previous step.

         .. code-block::

            atlas projects create myEphemeralProject --orgId <ORG-ID> --ownerId <YOUR-USER-ID>

      c. Retrieve the project ID for the ephemeral project you created
         from the response. In the following example response, the

>          project ID is 64933bde48add154124e343f.

Reviewer: I think using some sort of formatting here will help users by calling this info out
```suggestion
         project ID is ``64933bde48add154124e343f``.
```

         .. code-block::

            Project '64933bde48add154124e343f' created.

         Alternatively, you can use the :ref:`atlas-projects-list`
         command to find the project ID.
    
      d. Run the :ref:`atlas-quickstart` command to create an ``M10``

>          {+cluster+} and a database user. Replace <YOUR-PASSWORD> with a password for the database user and replace <PROJECT-ID> with

Reviewer: does this create a database user?

Author: Yes

         provides the :manual:`connection string 
         </reference/connection-string/#connection-string-options>` and
         connects to your {+cluster+} through {+mongosh+}.
         
   .. step:: Test your automations on the ephemeral project and {+cluster+}.

   .. step:: Delete the ephemeral project, {+cluster+}, and database user.

      a. Run the :ref:`atlas-clusters-delete` command to delete the
         ephemeral {+cluster+} and its database users. Replace <PROJECT-ID> with the project ID for your ephemeral project.

         .. code-block::

            atlas clusters delete myEphemeralCluster --projectId <PROJECT-ID> --force

      b. Run the :ref:`atlas-projects-delete` command to delete the
         ephemeral project. Replace <PROJECT-ID> with the project ID
         for your ephemeral project.

         .. code-block::

            atlas projects delete <PROJECT-ID> --force

         .. note::

            You can't delete the ephemeral project until the ephemeral
            {+cluster+} finishes shutting down. If you get an error
            stating ``CANNOT_CLOSE_GROUP_ACTIVE_ATLAS_CLUSTERS``, wait

>             a few minutes, then run the ``atlas projects delete``

Reviewer: `a few` is vague. Can we provide a specific recommendation here? Maybe five minutes?
```suggestion
            a few minutes, then run the ``atlas projects delete``
```
